src/fu-console.c

```diff
@@ -0,0 +1,753 @@
+/*
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuProgressBar"
+
+#include "config.h"
+
+#include <glib/gi18n.h>
+#include <stdio.h>
+
+#include "fu-console.h"
+
+#ifdef _WIN32
+#include <wchar.h>
+#include <windows.h>
+#endif
+
+struct _FuConsole {
+	GObject parent_instance;
+	GMainContext *main_ctx;
+	FwupdStatus status;
+	gboolean spinner_count_up; /* chars */
+	guint spinner_idx;	   /* chars */
+	guint length_percentage;   /* chars */
+	guint length_status;	   /* chars */
```

Comment:
I realize this was lifted from fu-progressbar.c, but what exactly is chars supposed to actually _mean_?

Reply:
it's the width in chars, which can be different to the number of bytes for UTF-8 output. I've added a comment.